bufferobserver.go

```diff
@@ -0,0 +1,13 @@
+package main
+
+// BufferObserver separates observer functionality out of file.
```

Comment:
I don't really agree that it does this. Something like: "BufferObserver implementations can register themselves with an ObserveableEditableBuffer instance so that it notifies them of  buffer mutations."?